hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestRootedOzoneFileSystem.java

```diff
@@ -1523,39 +1521,6 @@ public void testGetTrashRoots() throws IOException {
     Assert.assertTrue(volume1.setOwner(prevOwner));
   }
 
-  /**
-   * Check that  files are moved to trash since it is enabled by
-   * fs.rename(src, dst, options).
-   */
-  @Test
-  @Flaky({"HDDS-5819", "HDDS-6451"})
-  public void testRenameToTrashEnabled() throws IOException {
```

Comment:
Thanks @sadanand48 for fixing the discrepancy between .Trash paths for ofs and o3fs with this patch.  Overall, your changes look good with restructuring the `super.moveToTrash` (`hadoop TrashPolicyDefault`) call within the `TrashPolicyOzone` to make the trash paths the same.  

Just one thing with removed flaky test.  Can you add a check into the `testTrash` test to check for the existence of the key renamed to the `userTrashCurrent` **_prior_** to checking for purge?  Currently the test just checks that the key is purged and not in that path - https://github.com/apache/ozone/blob/5a69806aeadd44ab520e38c32240034728d55479/hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestRootedOzoneFileSystem.java#L1620